### ISSUE TITLE
Nextcloud 13.0.6 upgrades to wrong version.... (14.0.3) caused webdav issues...

### ISSUE THREAD

**Issue by noci2012** [nextcloud/server#11818]
Nextcloud 13.0.6 said an upgrade to 13.0.7 was present.
It also said upgrade to 14.x was NOT possible due to some modules.... no problem i use those modules.

Upgrade started, after restart: Nextcloud 14.0.3   - Really? 
Some modules were not working there known in advance, 
upgrade tool suggested upgrade to 13.0.7 was available...?

Well ok some modules could be working later... PDF reader doesn't though, Spreed.Me neither.

Also broken again: mounting a remote Alfresco: WebDav access.
I did have access in 13.0.6
In 14.0.3 the following errors appear:

Error	PHP Undefined offset: 1 at /3rdparty/sabre/dav/lib/DAV/Xml/Element/Response.php#237
Error	PHP	Undefined index: {DAV:}status at /3rdparty/sabre/dav/lib/DAV/Xml/Element/Response.php#236

Result: no files to be seen.
WebDav on Alfresco works using the native library or a Linux mount -t davfs

When configuring a new one after entering authentication info a green tick-mark is shown.
(The original entry from 13.0.6, also shows a green tick-mark).

I would not have expected to see old errors (NextCloud 10 era pop up again, 2016-ish).

Other observation: sometimes the selection of Files, then External Storage selection shows no entries, but does show the summary: 3 folders... (I have 3 configured now).  Reload fixes this.

**Comment by noci2012** [comment]
GitMate.io has it ALL wrong.

**Comment by noci2012** [comment]
Tried a litmus test...
All succeed except for:
copy_overwrite, copy_coll, copy_shallow, move,  move_coll

**Comment by MorrisJobke** [comment]
> PDF reader doesn't though, Spreed.Me neither.

What is the specific problem with the PDF reader? Please report it then to https://github.com/nextcloud/files_pdfreader.

Spreed.me is discontinued. You should switch to the successor "Talk", which is actively maintained and working fine on 14.



> Error	PHP Undefined offset: 1 at /3rdparty/sabre/dav/lib/DAV/Xml/Element/Response.php#237
> Error	PHP	Undefined index: {DAV:}status at /3rdparty/sabre/dav/lib/DAV/Xml/Element/Response.php#236

This one looks weird. Could you check the log file in nextcloud.log in your data directory for this and create a separate ticket for this and maybe check if this is covered in #1228?

> Also broken again: mounting a remote Alfresco: WebDav access.

Could this be the same as #1228?




> Tried a litmus test...
> All succeed except for:
> copy_overwrite, copy_coll, copy_shallow, move, move_coll, lock_shared,

@rullzer Is this known?

**Comment by pieter-groeneweg** [comment]
Same as in first post. Update says 13.0.7 available.. but when process started it says it will update to 14.0.3...
I stopped the process...

**Comment by danielkesselberg** [comment]
> Same as in first post. Update says 13.0.7 available.. but when process started it says it will update to 14.0.3...
> I stopped the process...

adding `'updater.release.channel' => 'production',` to config.php did the trick for me (https://nextcloud.com/blog/how-to-pick-the-right-nextcloud-releases-channel/)

**Comment by pieter-groeneweg** [comment]
Great, thanks... I think the description in that page needs some explanation on the versioning then... ;)

**Comment by noci2012** [comment]
PDF reader cannot be enabled on 14. So it doesn;t work by definition... didn't try to hack it to work.

I have Stable as release channel.
The updater showed it Couldn't update to 14 as there were dependencies not fullfilled.
So an upgrade to 13.0.7 should have worked as such..., so why does to updater Block upgrade to 14, but does upgrade to 14 anyway when there becomes a new 13 release available.

**Comment by noci2012** [comment]
> Could this be the same as #1228?
> 
> > Tried a litmus test...
> > All succeed except for:
> > copy_overwrite, copy_coll, copy_shallow, move, move_coll, lock_shared,

127.0.0.1 - - [16/Oct/2018:18:14:31 +0200] "PROPFIND /alfresco/webdav/ HTTP/1.0" 401 -
127.0.0.1 - - [16/Oct/2018:18:14:31 +0200] "PROPFIND /alfresco/webdav/ HTTP/1.0" 207 767
127.0.0.1 - - [16/Oct/2018:18:14:31 +0200] "PROPFIND /alfresco/webdav/ HTTP/1.0" 401 -
127.0.0.1 - - [16/Oct/2018:18:14:31 +0200] "PROPFIND /alfresco/webdav/ HTTP/1.0" 207 767
127.0.0.1 - - [16/Oct/2018:18:14:31 +0200] "PROPFIND /alfresco/webdav/ HTTP/1.0" 207 401
127.0.0.1 - - [16/Oct/2018:18:14:31 +0200] "PROPFIND /alfresco/webdav/ HTTP/1.0" 207 401

It concerns Alfresco 4.2f and yes it seems to be.  (not i got directory listings and could traverse them in 13.0.6)

**Comment by MorrisJobke** [comment]
> Same as in first post. Update says 13.0.7 available.. but when process started it says it will update to 14.0.3...

Currently we don't have a way to lockdown the version. The web UI sends some more information and thus can get a different version. The updater itself will always get the latest version on the channel. It's a bit tricky and something like https://github.com/nextcloud/updater/issues/189 would help to avoid this issue in the future. Nevertheless this only happens when we do staged rollouts, because then the web UI in the settings will maybe get into a different pool than the updater.

And yes the change to production works, because there we don't yet ship 14, but will soon do so AFAIK.

**Comment by pieter-groeneweg** [comment]
Wouldn't it be an idea then to have a checkbox that allows or denies a next major version?

**Comment by MorrisJobke** [comment]
> Wouldn't it be an idea then to have a checkbox that allows or denies a next major version?

We will find a way, but usually more config options are not the way we want to go. For this we have already the release channels.

**Comment by MorrisJobke** [comment]
> I understand.. yes, but from 13.0.6 to 13.0.7 I consider an update whereas from 13 to 14 I consider upgrade. Should that not be a part of the choice? Checkbox for upgrade.. I can see you do not want extra config options, but since there is a choice to make, there simply is an extra option.
> 
> An admin needs to be able to make a choice between updating or upgrading.

We don't have this concept yet. Also we use the term "updater" for the tool that just replaces old code with new code and only this. And "occ upgrade" to actually trigger the migrations once the new code is in place. Just to make this a bit more clear.

**Comment by MorrisJobke** [comment]
The alfresco issue is a duplicate of #1228 -> closing
